pallets/pooled-rewards/src/mock.rs

```diff
@@ -0,0 +1,130 @@
+use frame_support::{parameter_types, traits::Everything};
+use sp_arithmetic::FixedI128;
+use sp_core::H256;
+use sp_runtime::{
+	testing::Header,
+	traits::{BlakeTwo256, IdentityLookup},
+};
+
+pub use currency::testing_constants::{
+	DEFAULT_COLLATERAL_CURRENCY, DEFAULT_NATIVE_CURRENCY, DEFAULT_WRAPPED_CURRENCY,
+};
+pub use primitives::{CurrencyId, VaultCurrencyPair, VaultId};
+
+use crate as pooled_rewards;
+use crate::{Config, Error};
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
+type Block = frame_system::mocking::MockBlock<Test>;
+
+// Configure a mock runtime to test the pallet.
+frame_support::construct_runtime!(
+	pub enum Test where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic,
+	{
+		System: frame_system::{Pallet, Call, Storage, Config, Event<T>},
+		Reward: pooled_rewards::{Pallet, Call, Storage, Event<T>},
+	}
+);
+
+pub type AccountId = u64;
+pub type BlockNumber = u64;
+pub type Index = u64;
+pub type SignedFixedPoint = FixedI128;
+
+parameter_types! {
+	pub const BlockHashCount: u64 = 250;
+	pub const SS58Prefix: u8 = 42;
+	pub const MaxRewardCurrencies: u32= 10;
+}
+
+pub const DEFAULT_WRAPPED_CURRENCY2: CurrencyId = CurrencyId::AlphaNum4(
+	*b"USDT",
+	[
+		20, 209, 150, 49, 176, 55, 23, 217, 171, 154, 54, 110, 16, 50, 30, 226, 102, 231, 46, 199,
+		108, 171, 97, 144, 240, 161, 51, 109, 72, 34, 159, 139,
+	],
+);
+pub const DEFAULT_WRAPPED_CURRENCY3: CurrencyId = CurrencyId::AlphaNum4(
+	*b"MXN\0",
+	[
+		20, 209, 150, 49, 176, 55, 23, 217, 171, 154, 54, 110, 16, 50, 30, 226, 102, 231, 46, 199,
+		108, 171, 97, 144, 240, 161, 51, 109, 72, 34, 159, 139,
+	],
+);
+
+pub const DEFAULT_WRAPPED_CURRENCY4: CurrencyId = CurrencyId::AlphaNum4(
```

Comment:
Nit: Can we change the names to be `...CURRENCY_{INDEX}`?

Reply:
Why doesn't it start at 1 though?

Reply:
I think there is already a DEFAULT_WRAPPED_CURRENCY defined elsewhere (which would be 1), but I will update the indexes to `_x`

Reply:
I will move them all up to testing constants in currency so they are all in the same place.